dashboard/app/jobs.go

```diff
@@ -230,15 +230,39 @@ func getNextJob(c context.Context, testManagers, bisectManagers map[string]bool)
 
 func createBisectJob(c context.Context, managers map[string]bool, reproLevel dashapi.ReproLevel) (
 	*Job, *db.Key, error) {
+	job, jobKey, err := findBugsForBisection(c, managers, reproLevel, JobBisectCause)
+	if job != nil || err != nil {
+		return job, jobKey, err
+	}
+	return findBugsForBisection(c, managers, reproLevel, JobBisectFix)
+}
+
+func findBugsForBisection(c context.Context, managers map[string]bool, reproLevel dashapi.ReproLevel, jobType JobType) (
+	*Job, *db.Key, error) {
+	bisectQuery := map[JobType]string{
```

Comment:
I think the following is easier to understand because it requires reading just 1 branch with all concrete values in place. Whereas with the current code one needs to read the query the 3 separate maps and select corresponding values from maps and combine it all into the final query to understand what exactly query is executed.
```
	// Note: we could also include len(Commits)==0 but datastore does not work this way.
	// So we would need an additional HasCommits field or something.
	// Note: For JobBisectCause, order the bugs from newest to oldest. For JobBisectFix,
	// order the bugs from oldest to newest.
	// Sort property should be the same as property used in the inequality filter.
	query := db.NewQuery("Bug").Filter("Status=", BugStatusOpen)
	if jobType == JobBisectCause {
		query = query.Filter("FirstTime>", time.Time{}).
			Filter("ReproLevel=", reproLevel).
			Filter("BisectCause=", BisectNot).
			Order("-FirstTime")
	} else {
		query = query.Filter("LastTime>", time.Time{}).
			Filter("ReproLevel=", reproLevel).
			Filter("BisectFix=", BisectNot).
			Order("LastTime")
	}
	// We only need 1 job, but we skip some because the query is not precise.
	var bugs []*Bug
	keys, err := query.Limit(300).GetAll(c, &bugs)
```
Please send a follow up change.

Reply:
Thanks! I've updated the PR.

Reply:
This is already merged, please send a separate PR.

Reply:
Done, I've created a PR here: https://github.com/google/syzkaller/pull/1304